### PR TITLE
US-115 | feat(graphql): type graphql search index with an enumeration for clarity

### PR DESCRIPTION
**Pull request by karisal-anders**
### feat(graphql): type graphql search index with an enumeration for clarity

also:
 - add "typecheck" command to graphql package.json for checking
   Typescript types
 - add more typings to the use of indices

refs US-115

## Tickets

[US-115](https://helsinkisolutionoffice.atlassian.net/browse/US-115)

[US-115]: https://helsinkisolutionoffice.atlassian.net/browse/US-115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr116.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr116.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr116.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**Unified search sources branch is deployed to platta: https://kuva-unified-search-sources-pr116.api.dev.hel.ninja** :rocket::rocket::rocket:

File: graphql/src/datasources/es.ts [hunk 1]
```diff
@@ -225,25 +225,25 @@ class ElasticSearchAPI extends RESTDataSource {
...
     const searchFieldsBoostMapping = {
       // Normally weighted search fields for different indexes
-      1: (lang: string, index: string) => {
+      1: (lang: ElasticLanguage, index: ElasticSearchIndex) => {
         return (
           {
-            location: [`venue.description.${lang}`],
-            event: [`event.name.${lang}`, `event.description.${lang}`],
+            ES_LOCATION_INDEX: [`venue.description.${lang}`],
+            ES_EVENT_INDEX: [`event.name.${lang}`, `event.description.${lang}`],
```
[the remark this answers is not in the document]
Comment (karisal-anders) [reply graphql/src/datasources/es.ts]: Ah, fixed with [force push](https://github.com/City-of-Helsinki/unified-search/compare/d4ddaaead74f442c2d71e4ce527c2ac3396c8ec8..e9994174dc77dccf64394845fe86a2d10f7bab50)

File: graphql/src/datasources/es.ts [hunk 2]
```diff
@@ -225,25 +225,25 @@ class ElasticSearchAPI extends RESTDataSource {
...
           }[index] ?? []
         );
       },
-      3: (lang: string, index: string) => {
+      3: (lang: ElasticLanguage, index: ElasticSearchIndex) => {
         return (
           {
-            location: [`venue.name.${lang}`],
+            ES_LOCATION_INDEX: [`venue.name.${lang}`],
```
[the remark this answers is not in the document]
Comment (karisal-anders) [reply graphql/src/datasources/es.ts]: Ah, fixed with [force push](https://github.com/City-of-Helsinki/unified-search/compare/d4ddaaead74f442c2d71e4ce527c2ac3396c8ec8..e9994174dc77dccf64394845fe86a2d10f7bab50)

**Comment by terovirtanen** [comment]
**Unified search graphql branch is deployed to platta: https://kuva-unified-search-pr116.api.dev.hel.ninja** :rocket::rocket::rocket:

**Review by nikomakela — APPROVED** [review]

**Pull request merged**